libraries/chain/controller.cpp

```diff
@@ -933,32 +933,6 @@ struct controller_impl {
 # warning TODO: Revisit this and adpat to new design
          if constexpr (std::is_same_v<value_t, kv_object>) {
             snapshot->write_section<value_t>([this]( auto& section ){
-               // This ordering depends on the fact the eosio.kvdisk is before eosio.kvram and only eosio.kvdisk can be stored in rocksdb.
-               if (db.get<kv_db_config_object>().using_rocksdb_for_disk) {
-                  std::unique_ptr<rocksdb::Iterator> it{kv_database.rdb->NewIterator(rocksdb::ReadOptions())};
-                  std::vector<char> prefix = rocksdb_contract_kv_prefix;
-                  b1::chain_kv::append_key(prefix, kvdisk_id.to_uint64_t());
-                  it->Seek(b1::chain_kv::to_slice(rocksdb_contract_kv_prefix));
-                  while(it->Valid()) {
-                     auto key = it->key();
-                     if(key.size() < rocksdb_contract_kv_prefix.size() || !std::equal(prefix.begin(), prefix.end(), key.data()))
-                        break;
-                     uint64_t contract;
-                     char buf[sizeof(contract)];
-                     std::size_t key_prefix_size = prefix.size() + sizeof(contract);
-                     EOS_ASSERT(key.size() >= key_prefix_size, database_exception, "Unexpected key in rocksdb");
-                     std::reverse_copy(key.data() + prefix.size(), key.data() + key_prefix_size, buf);
-                     std::memcpy(&contract, buf, sizeof(contract));
-                     auto value = it->value();
-                     kv_object_view row{
-                        name(contract),
-                        { { key.data() + key_prefix_size, key.data() + key.size() } },
-                        { { value.data(), value.data() + value.size() } }
-                     };
-                     section.add_row(row, db);
-                     it->Next();
-                  }
-               }
```

Comment:
I always had trouble understanding this section of code. With disk removed, we don't need this code any more?

Reply:
That was my understanding of it given that `using_rocksdb_for_disk` isn't a concept anymore. But somebody more knowledgable of controller can chime in, I'm not 100% positive on that.

Reply:
This code is copying all KV values stored in RocksDB to the snapshot database. It shouldn't have been deleted. There's a similar piece in read_from_snapshot(). It also should be preserved.

Reply:
Ah I underestand, the naming had me a bit confused. I'll restore.